Convert to V2

Thovi98 opened the pull request:
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

Thovi98 commented:
!testme

Thovi98 commented:
!testme

Thovi98 commented:
!testme

alexAubin commented on scripts/change_url:
```diff
@@ -10,68 +10,11 @@ source _common.sh
-then
-	change_path=1
-fi
+ynh_app_setting_set --app=$app --key=language --value=$language
+ynh_app_setting_set --app=$app --key=crypto_random --value=$crypto_random
+ynh_app_setting_set --app=$app --key=web_api_password --value=$web_api_password
```
Bit confused about these lines, doesn't seem to achieve anything, you're setting variables to their current value ...

Thovi98 replied:
Thank you for this review!
Yes, I’ve added this because if I remember well, there was an error shown in the CI about WEB_API_PASSWORD not initialized (cf. https://ci-apps-dev.yunohost.org/ci/job/11195) and this seemed to fix it I think.

Anyway, I still have an issue with the `overleaf-history-v1.service` failing to start (I will soon try to install it again because I don’t remember the exact issue), so the app doesn’t work yet.

alexAubin replied:
Zblerg yes the web_api_password setting thing was not available in upgrade nor change_url, because you did not save it during install at that time, which you then fixed in https://github.com/YunoHost-Apps/overleaf_ynh/pull/21/commits/c6ba9b47433f3784bff4b7669ae73fa2aa795a44 (by adding the `ynh_settings_set` in the install script)

But definitely you don't need to write `foobar=foobar` in the upgrade and changeurl now that the setting is properly set

alexAubin commented on scripts/install:
```diff
@@ -7,85 +7,31 @@
 
-ynh_app_setting_set --app=$app --key=domain --value=$domain
-ynh_app_setting_set --app=$app --key=path --value=$path_url
 ynh_app_setting_set --app=$app --key=language --value=$language
```
```suggestion
```

already saved from the install questions

alexAubin commented on scripts/install:
```diff
@@ -7,85 +7,31 @@
-# Register (book) web path
-ynh_webpath_register --app=$app --domain=$domain --path_url=$path_url
-
 #=================================================
 # STORE SETTINGS FROM MANIFEST
 #=================================================
```
```suggestion
```

alexAubin commented on scripts/restore:
```diff
@@ -8,41 +8,29 @@
-}
-# Exit if an error occurs during the execution of the script
-ynh_abort_if_errors
+ynh_app_setting_set --app=$app --key=web_api_password --value=$web_api_password
```
```suggestion
```

??? x_x

alexAubin commented on scripts/upgrade:
```diff
@@ -7,28 +7,14 @@
-port=$(ynh_app_setting_get --app=$app --key=port)
-web_api_password=$(ynh_app_setting_get --app=$app --key=web_api_password)
-crypto_random=$(ynh_app_setting_get --app=$app --key=crypto_random)
+ynh_app_setting_set --app=$app --key=language --value=$language
+ynh_app_setting_set --app=$app --key=crypto_random --value=$crypto_random
+ynh_app_setting_set --app=$app --key=web_api_password --value=$web_api_password
```
```suggestion
```

butwhy.gif

alexAubin reviewed (commented):
LGTM apart from minor comments, but i'm a bit confused about all the nodejs warnings/(errors?) in the CI

Thovi98 commented:
!testme

cichy1173 commented:
Thank yo for your work @Thovi98 

Is the currently prepared package stable enough to use? Or have you fixed the upload of photos and other files?

Thovi98 commented:
> Is the currently prepared package stable enough to use?

No, I've an issue with the history-v1 service which keeps failing... the app installs well but doesn't work ^^ I still have to debug it more.

> have you fixed the upload of photos and other files?

I must say that I've never used the app before so I don't know what you are refering to 😁
If and when I find the solution, I will try to make the package with as few bugs as possible.

cichy1173 commented:
@Thovi98 

> I must say that I've never used the app before so I don't know what you are refering to 😁. If and when I find the solution, I will try to make the package with as few bugs as possible.

A few months ago I was able to install Overleaf_ynh and actually the application worked. Unfortunately, it was not possible to upload any files to projects. This made the platform basically useless. From what I was able to read at the time, this was a typical problem for all users.

Thovi98 commented:
For reference, the current (or one of I don’t know) issue is that `overleaf-history-v1.service` fails.

```
LOGFILE: journalctl
-- Journal begins at Sun 2023-12-17 04:35:33 CET, ends at Sun 2023-12-17 16:32:20 CET. --
déc 17 16:31:26 systemd[1]: overleaf-history-v1.service: Consumed 2.842s CPU time.
déc 17 16:31:31 systemd[1]: overleaf-history-v1.service: Scheduled restart job, restart counter is at 119.
déc 17 16:31:31 systemd[1]: Stopped overleaf history-v1.
déc 17 16:31:31 systemd[1]: overleaf-history-v1.service: Consumed 2.842s CPU time.
déc 17 16:31:31 systemd[1]: Started overleaf history-v1.
déc 17 16:31:34 systemd[1]: overleaf-history-v1.service: Main process exited, code=exited, status=1/FAILURE
déc 17 16:31:34 systemd[1]: overleaf-history-v1.service: Failed with result 'exit-code'.
déc 17 16:31:34 systemd[1]: overleaf-history-v1.service: Consumed 2.842s CPU time.
déc 17 16:31:39 systemd[1]: overleaf-history-v1.service: Scheduled restart job, restart counter is at 120.
déc 17 16:31:39 systemd[1]: Stopped overleaf history-v1.
déc 17 16:31:39 systemd[1]: overleaf-history-v1.service: Consumed 2.842s CPU time.
déc 17 16:31:39 systemd[1]: Started overleaf history-v1.
déc 17 16:31:41 systemd[1]: overleaf-history-v1.service: Main process exited, code=exited, status=1/FAILURE
déc 17 16:31:41 systemd[1]: overleaf-history-v1.service: Failed with result 'exit-code'.
déc 17 16:31:41 systemd[1]: overleaf-history-v1.service: Consumed 2.817s CPU time.
déc 17 16:31:46 systemd[1]: overleaf-history-v1.service: Scheduled restart job, restart counter is at 121.
déc 17 16:31:46 systemd[1]: Stopped overleaf history-v1.
déc 17 16:31:46 systemd[1]: overleaf-history-v1.service: Consumed 2.817s CPU time.
déc 17 16:31:46 systemd[1]: Started overleaf history-v1.
déc 17 16:31:49 systemd[1]: overleaf-history-v1.service: Main process exited, code=exited, status=1/FAILURE
déc 17 16:31:49 systemd[1]: overleaf-history-v1.service: Failed with result 'exit-code'.
déc 17 16:31:49 systemd[1]: overleaf-history-v1.service: Consumed 2.816s CPU time.
déc 17 16:31:54 systemd[1]: overleaf-history-v1.service: Scheduled restart job, restart counter is at 122.
déc 17 16:31:54 systemd[1]: Stopped overleaf history-v1.
déc 17 16:31:54 systemd[1]: overleaf-history-v1.service: Consumed 2.816s CPU time.
déc 17 16:31:54 systemd[1]: Started overleaf history-v1.
déc 17 16:31:56 systemd[1]: overleaf-history-v1.service: Main process exited, code=exited, status=1/FAILURE
déc 17 16:31:56 systemd[1]: overleaf-history-v1.service: Failed with result 'exit-code'.
déc 17 16:31:56 systemd[1]: overleaf-history-v1.service: Consumed 2.770s CPU time.
déc 17 16:32:02 systemd[1]: overleaf-history-v1.service: Scheduled restart job, restart counter is at 123.
déc 17 16:32:02 systemd[1]: Stopped overleaf history-v1.
déc 17 16:32:02 systemd[1]: overleaf-history-v1.service: Consumed 2.770s CPU time.
déc 17 16:32:02 systemd[1]: Started overleaf history-v1.
déc 17 16:32:04 systemd[1]: overleaf-history-v1.service: Main process exited, code=exited, status=1/FAILURE
déc 17 16:32:04 systemd[1]: overleaf-history-v1.service: Failed with result 'exit-code'.
déc 17 16:32:04 systemd[1]: overleaf-history-v1.service: Consumed 2.775s CPU time.
déc 17 16:32:09 systemd[1]: overleaf-history-v1.service: Scheduled restart job, restart counter is at 124.
déc 17 16:32:09 systemd[1]: Stopped overleaf history-v1.
déc 17 16:32:09 systemd[1]: overleaf-history-v1.service: Consumed 2.775s CPU time.
déc 17 16:32:09 systemd[1]: Started overleaf history-v1.
déc 17 16:32:12 systemd[1]: overleaf-history-v1.service: Main process exited, code=exited, status=1/FAILURE
déc 17 16:32:12 systemd[1]: overleaf-history-v1.service: Failed with result 'exit-code'.
déc 17 16:32:12 systemd[1]: overleaf-history-v1.service: Consumed 2.749s CPU time.
déc 17 16:32:17 systemd[1]: overleaf-history-v1.service: Scheduled restart job, restart counter is at 125.
déc 17 16:32:17 systemd[1]: Stopped overleaf history-v1.
déc 17 16:32:17 systemd[1]: overleaf-history-v1.service: Consumed 2.749s CPU time.
déc 17 16:32:17 systemd[1]: Started overleaf history-v1.
déc 17 16:32:19 systemd[1]: overleaf-history-v1.service: Main process exited, code=exited, status=1/FAILURE
déc 17 16:32:19 systemd[1]: overleaf-history-v1.service: Failed with result 'exit-code'.
déc 17 16:32:19 systemd[1]: overleaf-history-v1.service: Consumed 2.796s CPU time.

LOGFILE: /var/log/overleaf/history-v1.log
Initializing metrics
Set UV_THREADPOOL_SIZE=16
/var/www/overleaf/live/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^

TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/var/www/overleaf/live/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at new ConnectionString (/var/www/overleaf/live/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/var/www/overleaf/live/services/history-v1/node_modules/mongodb/lib/connection_string.js:185:17)
    at new MongoClient (/var/www/overleaf/live/services/history-v1/node_modules/mongodb/lib/mongo_client.js:51:63)
    at Object.<anonymous> (/var/www/overleaf/live/services/history-v1/storage/lib/mongodb.js:6:16)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:119:18)
    at Object.<anonymous> (/var/www/overleaf/live/services/history-v1/storage/lib/blob_store/index.js:17:17)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)

Node.js v18.18.2
Initializing metrics
Set UV_THREADPOOL_SIZE=16
/var/www/overleaf/live/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^

TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/var/www/overleaf/live/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at new ConnectionString (/var/www/overleaf/live/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/var/www/overleaf/live/services/history-v1/node_modules/mongodb/lib/connection_string.js:185:17)
    at new MongoClient (/var/www/overleaf/live/services/history-v1/node_modules/mongodb/lib/mongo_client.js:51:63)
    at Object.<anonymous> (/var/www/overleaf/live/services/history-v1/storage/lib/mongodb.js:6:16)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:119:18)
    at Object.<anonymous> (/var/www/overleaf/live/services/history-v1/storage/lib/blob_store/index.js:17:17)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)

Node.js v18.18.2
```

Thovi98 commented:
!testme

Thovi98 commented:
!testme

ericgaspar commented on manifest.toml:
```diff
@@ -0,0 +1,83 @@
+
+    [resources.install_dir]
+
+    [resources.data_dir]
+
```
```suggestion
    [resources.data_dir]
    subdirs = ["compiles", "output", "cache"]
```

Thovi98 replied:
Yeah thanks!

ericgaspar commented on scripts/install:
```diff
@@ -126,84 +66,90 @@ ynh_add_nginx_config
 #=================================================
 ynh_script_progression --message="Creating a data directory..."
 
-datadir=/home/yunohost.app/$app
```
we can remove this `CREATE DATA DIRECTORY` section (we creat sudirectories in the `manifest.toml` file)

Thovi98 commented:
!bookwormtestme

Thovi98 commented:
!testme

Thovi98 commented:
![image](https://github.com/YunoHost-Apps/overleaf_ynh/assets/51749973/361accaf-12e2-47c4-9c72-215780b63152)

Yeaaah!
There are still some fixes to be done (like but it’s in a good way 🎉

Thovi98 commented:
!testme

cichy1173 commented:
> ![image](https://private-user-images.githubusercontent.com/51749973/303321755-361accaf-12e2-47c4-9c72-215780b63152.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc1NzUwNTYsIm5iZiI6MTcwNzU3NDc1NiwicGF0aCI6Ii81MTc0OTk3My8zMDMzMjE3NTUtMzYxYWNjYWYtMTJlMi00N2M0LTljNzItMjE1NzgwYjYzMTUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjEwVDE0MTkxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3ZTI1NzIyZDgwYjkxNzEyZmY5OTFjMGEwNWFhYmQ3M2Q2NTRjZDI5NzE3MWM3MzIwNmEwZDkxZDRjNTFlNTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UKb6ljb9yCy1Fbg9wLpRQssa51TNeJSr99zkRJniJhs)
> 
> Yeaaah! There are still some fixes to be done (like LDAP) but it’s in a good way 🎉

Oh my god! I am terribly waiting for the final version! Overleaf is problematic on many platforms, including Docker. If it works on yunohost then I'll be so happy!

ericgaspar commented on scripts/install:
```diff
@@ -126,7 +126,7 @@ ynh_script_progression --message="Building app..."
 	ynh_use_nodejs
 	#ynh_exec_warn_less $ynh_npm install sandboxed-module@latest
-    	npm ci --quiet
+    	ynh_exec_warn_less npm ci
```
```suggestion
    	exec_warn_less ynh_exec_as $app env $ynh_node_load_PATH $ynh_npm ci
```

ericgaspar replied:
maybe...

Thovi98 replied:
Why would it be better? :)

Thovi98 commented:
!testme

Thovi98 commented:
!testme

Thovi98 commented:
!testme

Thovi98 commented:
Okay, from my point it’s ready now!

Some observations:
- I did not try to upgrade from the latest working version (because it’s not working anymore). So I don’t garantee it will successfully upgrade (nor that your actual projects will successfully migrate).
- I removed LDAP integration because it was difficult enough to make the package work already. BUT emails should work (tested on my end).
- @cichy1173 talked about an upload issue. I found the reason of it and fixed it (nginx config issue)
- It’s working _for now_. I don’t garantee new versions will be easily upgradable.

cichy1173 commented:
Great! Waiting for master release
